Fix Gentoo Bug 268592, patch adapted from upstream patch.

Modified to apply to a release tarball (no libsoup-2.4-docs.sgml.in)
Hence, to be applied ONLY when !gnome && doc

---
--- docs/reference/Makefile.am
+++ docs/reference/Makefile.am
@@ -13,7 +13,7 @@
 DOC_SOURCE_DIR=../../libsoup
 
 # Extra options to supply to gtkdoc-scan.
-SCAN_OPTIONS=--deprecated-guards=LIBSOUP_DISABLE_DEPRECATED
+SCAN_OPTIONS=--deprecated-guards=LIBSOUP_DISABLE_DEPRECATED --rebuild-types
 
 # Extra options to supply to gtkdoc-scangobj.
 SCANGOBJ_OPTIONS=--type-init-func 'g_type_init(); g_thread_init(NULL)'
@@ -38,6 +38,9 @@
 	soup-proxy-resolver-libproxy.h soup-proxy-resolver-gconf.h \
 	soup-proxy-resolver-static.h
 
+IGNORE_HFILES += soup-cookie-jar-sqlite.h \
+	soup-gnome.h soup-gnome-features.h
+
 # Images to copy into HTML directory.
 HTML_IMAGES = 
 
@@ -61,7 +61,7 @@
 	$(GNUTLS_CFLAGS)		\
 	$(LIBPROXY_LIBS)
 
-GTKDOC_LIBS = $(top_builddir)/libsoup/libsoup-gnome-2.4.la
+GTKDOC_LIBS = $(top_builddir)/libsoup/libsoup-2.4.la
 
 # include common portion ...
 include $(top_srcdir)/gtk-doc.make
--- docs/reference/libsoup-2.4-docs.sgml
+++ docs/reference/libsoup-2.4-docs.sgml
@@ -47,11 +47,13 @@
     <xi:include href="xml/soup-value-utils.xml"/>
   </chapter>
 
+  <!--
   <chapter>
     <title>GNOME integration</title>
     <xi:include href="xml/soup-gnome-features.xml"/>
     <xi:include href="xml/soup-cookie-jar-sqlite.xml"/>
   </chapter>
+  -->
 
   <chapter>
     <title>Low-level Networking API</title>
